File: DSA/LeetCode/Stack/2816-double-a-number-represented-as-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode doubleIt(ListNode head) {

        Stack<ListNode> stack = new Stack<ListNode>();

        ListNode curr = head;
        while (curr != null) {
            stack.push(curr);
            curr = curr.next;
        }

        int acc = 0;
        while (!stack.isEmpty()) {
            ListNode node = stack.pop();
            int val = (node.val * 2) + acc;
            node.val = val % 10;
            acc = val / 10;
        }

        return (acc > 0) ? new ListNode(acc, head) : head;
    }
}

File: DSA/LeetCode/Stack/2816-double-a-number-represented-as-a-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def doubleIt(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        stack = []

        curr = head
        while curr:
            stack.append(curr)
            curr = curr.next
        

        acc = 0
        while  stack: 
             node = stack.pop()
             val = (node.val * 2) + acc
             acc, node.val =  divmod(val , 10)
            
        return ListNode(acc, head) if acc > 0 else head
        # return (acc > 0) ? new ListNode(acc, head) : head
        